controllers/tenant/apimanagementtenant_controller.go

```diff
@@ -136,9 +136,11 @@ func (r *TenantReconciler) getAPIManagementTenant(crName string, crNamespace str
 	return tenant, nil
 }
 
+// The purpose of this method is to verify if particular APIManagementTenant is created or in progress of creation. If yes then skip, if no then proceed with checking if there is ANOTHER APIManagementTenant already created or in progress of creation
+
 func (r *TenantReconciler) verifyAPIManagementTenant(tenant *v1alpha1.APIManagementTenant) (bool, string, error) {
 	// Skip verification if the tenant is already reconciled
-	if tenant.Status.ProvisioningStatus != v1alpha1.ThreeScaleAccountReady {
+	if tenant.Status.ProvisioningStatus != v1alpha1.ThreeScaleAccountReady && tenant.Status.ProvisioningStatus != v1alpha1.ThreeScaleAccountRequested && tenant.Status.ProvisioningStatus != v1alpha1.UserAnnotated {
```

Comment:
Given that there are only four options here: https://github.com/integr8ly/integreatly-operator/blob/master/apis/v1alpha1/apimanagementtenant_types.go#L25-L30 would not it be much simpler to change the condition to `tenant.Status.ProvisioningStatus == v1alpha1.WontProvisionTenant`

However, it does not make sense to me. I might be missing something here probably

Reply:
Probably not good enough since a situation when `ProvisioningStatus` was not yet populated needs to be addressed as well?

Reply:
Thanks @carlkyrillos so `tenant.Status.ProvisioningStatus == v1alpha1.WontProvisionTenant || tenant.Status.ProvisioningStatus == ""` might potentially work too but that's probably not worth the effort to change and retest.

Reply:
@Stellar-9 @trepel I actually think this line should be reverted to only skip verifying an APIManagementTenant CR if it's already been reconciled (and therefore already verified). I don't think there's any harm in continuously verifying all other CRs.
```suggestion
	if tenant.Status.ProvisioningStatus != v1alpha1.ThreeScaleAccountReady{
```

Reply:
I am not very familiar with the logic to comment on that. It was just that long condition that caught my eye

Reply:
@trepel  I think my if statement is more conscise and easier to read. But yours is more explicit and less error-prone maybe. But I think it's matter of taste mostly

@carlkyrillos So I would be happy if you can correct me if I'm wrong. But I think that might be not enough for what we want to achieve based on that JIRA. If user will create CR and provision status will not be 3scale account ready yet, but some of the other ones. So still in process of creation and user will try to create another CR, then in that case it will proceed, which we don't want to allow them. I mean in testing enviroment, CR creation is pretty quick and almost imidiately in status 3scale account ready. But based on that JIRA was created, I guess in production are actually this conflict situations happening. And from my perspectie I don't see any cons to have this statement there, just in case. But I don't know. if I'm wrong here, that I would be happy to learn more about it

Reply:
@Stellar-9 Let's chat offline to discuss further but I think changing [this line](https://github.com/integr8ly/integreatly-operator/blob/c431f386775a5f8a7d5c828d6c787d26c210e5f6/controllers/tenant/apimanagementtenant_controller.go#L143) to:
`if tenant.Status.ProvisioningStatus == ""`
and changing [this line](https://github.com/integr8ly/integreatly-operator/blob/c431f386775a5f8a7d5c828d6c787d26c210e5f6/controllers/tenant/apimanagementtenant_controller.go#L171) to:
`if t != tenant && t.Status.ProvisioningStatus != v1alpha1.WontProvisionTenant`
will prevent the conflict and simplify the `if` statements but I will need to test to confirm.

Reply:
@Stellar-9 @trepel Sorry for all of my flip-flopping but after testing/verifying a few different options, I think Ales' original implementation is best and the PR can be merged as is.

Reply:
Thanks for clarifications, guys! I am fine with this as is. It's approved now anyway.